bug: flaky TestACPConformance/Stop_ConcurrentDistinctSessions

### Before you continue

- [x] I searched existing issues and did not find a duplicate.
- [x] I read the relevant docs and contributor guidance.

### Gas City version

main@561881caa5 (CI run on 2026-04-09)

### Environment

GitHub Actions CI runner (ubuntu, `make test-cover`)

### Reproduction

Observed intermittently in CI. The test is in `internal/runtime/acp`:

```
go test ./internal/runtime/acp/...
```

The failure is non-deterministic — it depends on goroutine scheduling during concurrent session stops.

### Expected behavior

`TestACPConformance/Stop_ConcurrentDistinctSessions` should pass — after concurrently stopping distinct sessions, `IsRunning()` should return `false` for all stopped sessions.

### Actual behavior

```
--- FAIL: TestACPConformance (0.37s)
    --- FAIL: TestACPConformance/Stop_ConcurrentDistinctSessions (0.01s)
        conformance.go:168: IsRunning(gc-acp-conform-10) = true after concurrent Stop
```

`IsRunning` returns `true` for a session that was concurrently stopped, suggesting a race between `Stop` completing and `IsRunning` reading state.

### Logs, screenshots, or traces

Full CI run: https://github.com/gastownhall/gascity/actions/runs/24200711973/job/70643193475

Observed during CI for PR #531 (unrelated change — reconciler fix for named-session `scale_check`). All other tests pass.

### Additional context

The test stops multiple distinct sessions concurrently and then asserts `IsRunning` is false for each. The 0.01s duration suggests the race window is tight. Possible causes:
- Missing synchronization between `Stop` goroutine completion and state query
- `IsRunning` reading stale state before the stop has fully propagated

--- a/internal/runtime/acp/acp.go
+++ b/internal/runtime/acp/acp.go
@@ -246,13 +246,19 @@ func (p *Provider) Start(ctx context.Context, name string, cfg runtime.Config) e
 	go sc.readLoop(stdoutPipe)
 
 	// Monitor process exit — clean up pending state, socket, and listener.
+	// Socket cleanup happens BEFORE close(done) so that callers waiting
+	// on sc.done (e.g., terminateProcess) can rely on the socket being
+	// gone when done fires. Without this ordering, IsRunning can race:
+	// Stop deletes the conn from the map, terminateProcess waits on done,
+	// done closes, Stop returns — but the socket is still alive, so
+	// IsRunning falls through to socketAlive and returns true.
 	go func() {
 		_ = cmd.Wait()
 		sc.drainPending()
-		close(sc.done)
 		lis.Close()                 //nolint:errcheck
 		os.Remove(p.sockPath(name)) //nolint:errcheck
 		_ = os.Remove(p.sockNamePath(name))
+		close(sc.done)
 	}()
 
 	// Perform ACP handshake with a deadline. hsCtx (created above with

